ext/mbstring/mbstring.c

```diff
@@ -334,7 +334,8 @@ static const mbfl_encoding *php_mb_get_encoding(zend_string *encoding_name) {
 
 		encoding = mbfl_name2encoding(ZSTR_VAL(encoding_name));
 		if (!encoding) {
-			php_error_docref(NULL, E_WARNING, "Unknown encoding \"%s\"", ZSTR_VAL(encoding_name));
+			zend_argument_value_error(arg_num, "must be a valid encoding, encoding \"%s\" is unknown",
```

Comment:
What about "must be a valid encoding format, %s given"? It sounds better for me. (Although i'm not sure if we still have to print the given value, since the argument name and order is displayed in the message)

Reply:
It's not really a format thought, "character set" could also work but the message is currently suboptimal I agree.

Reply:
I'd prefer the middle ground between those, which is `must be a valid encoding, "%s" given`.